index.js

```diff
@@ -0,0 +1,306 @@
+function popupBlocker(window) {
+
+/************************************************************************************/
+var top = window.top;
+var PROP = typeof Symbol == 'function' ? Symbol() : Math.random().toString(36).substr(7);
+var MouseEvent = top.MouseEvent;
+/************************************************************************************/
+// Verify whether the current event handler that is being executed
+// is artificially created to generate popups.
+var currentEvent;
+var retrieveEvent = function() {
+    log('Retrieving event');
+    currentEvent = top.event;
+    if (!currentEvent) {
+        log('window.event does not exist, trying to get event from Function.caller');
+        try {
+            var caller = arguments.callee;
+            while (caller) { caller = caller.caller; }
+            log('Reached at the top of caller chain.');
+            if (caller.arguments && caller.arguments[0] && caller.arguments[0].target) {
+                currentEvent = caller.arguments[0];
+                log('The function at the bottom of the stack has an expected type. The current event is:', currentEvent);
+            }
+        } catch (e) {
+            log('Getting event from Function.caller failed, due to an error:', e);
+        }
+    } else {
+        log('window.event exists, of which the value is:', currentEvent);
+    }
+};
+
+/**
+ * @param {*} event Optional argument, an event to test with. Default value is currentEvent.
+ * @return {boolean} True if the event is legit, false if it is something that we should not allow window.open or dispatchEvent.
+ */
+var verifyEvent = function(event) {
```

Comment:
This method also looks simple enough to be tested.

Tests should look more or less like this:
```
var event = ... creating test event object ...
assertTrue(verifyEvent(event));
```